skopt/searchcv.py

```diff
@@ -0,0 +1,448 @@
+
+    """
+
+    def __init__(self, estimator, search_spaces, optimizer_kwargs=None,
+                 n_iter=256, scoring=None, fit_params=None, n_jobs=1,
+                 iid=True, refit=True, cv=None, verbose=0,
+                 pre_dispatch='2*n_jobs', random_state=None,
+                 error_score='raise', return_train_score=True):
+        self.search_spaces = search_spaces
+        self.n_iter = n_iter
+        self.random_state = random_state
+
+        if optimizer_kwargs is None:
+            self.optimizer_kwargs = {}
+        else:
+            self.optimizer_kwargs = optimizer_kwargs
+
+        # this dict is used in order to keep track of skopt Optimizer
+        # instances for different search spaces (str(space) is used as key)
+        self.optimizer_ = {}
+        self.cv_results_ = defaultdict(list)
+
+        self.best_index_ = None
+
+        super(BayesSearchCV, self).__init__(
+             estimator=estimator, scoring=scoring, fit_params=fit_params,
+             n_jobs=n_jobs, iid=iid, refit=refit, cv=cv, verbose=verbose,
+             pre_dispatch=pre_dispatch, error_score=error_score,
+             return_train_score=return_train_score)
+
+    def _fit_best_model(self, X, y):
+        """Fits the estimator copy with best parameters found to the
+        provided data.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Input data, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples] or [n_samples, n_output],
+            Target relative to X for classification or regression.
+
+        Returns
+        -------
+        self
+        """
+        self.best_estimator_ = clone(self.estimator)
+        self.best_estimator_.set_params(**self.best_params_)
+        self.best_estimator_.fit(X, y)
+        return self
+
+    def _make_optimizer(self, params_space):
+        """Method to instantiate skopt Optimizer class.
+
+        Parameters
+        ----------
+        params_space : dict
+            Represents parameter search space. The keys are parameter
+            names (strings) and values are skopt.space.Dimension instances,
+            one of Real, Integer or Categorical.
+
+        Returns
+        -------
+        optimizer: Instance of the `Optimizer` class used for for search
+            in some parameter space.
+
+        """
+        # convert search space from dict to list
+        dimensions = [params_space[k] for k in sorted(params_space.keys())]
+
+        kwargs = self.optimizer_kwargs.copy()
+        kwargs['dimensions'] = dimensions
+        optimizer = Optimizer(**kwargs)
+
+        return optimizer
+
+    def _skopt_to_dict(self, params_space, params):
+        """Converts list of parameter values into the dictionary with
+        keys as parameter names and corresponding values of parameter.
```

Comment:
Would change to:
"""
Convert a list of parameter values into a dictionary.

Parameter names are used as the key for that parameter.
...
"""